### ISSUE TITLE
How to setup multiple databases and vhosts in blt

### ISSUE THREAD

**Issue by retrodans** [acquia/blt#2455]
My system information:
* Operating system type: Ubuntu
* Operating system version: 16.04
* BLT version: 8.9.12

Output of `blt doctor`:
```
+----------------------+----------------------------------------------------------------------------------+
| Property             | Value                                                                            |
+----------------------+----------------------------------------------------------------------------------+
| %paths               |                                                                                  |
|  - %root             | /var/www/gigaclear/docroot                                                       |
|  - %site             | sites/default                                                                    |
|  - %modules          | modules                                                                          |
|  - %themes           | themes                                                                           |
|  - %config-sync      | /var/www/gigaclear/config/default                                                |
|  - %files            | sites/default/files                                                              |
|  - %private          | /var/www/gigaclear/files-private                                                 |
|  - %temp             | /tmp                                                                             |
| admin-theme          | seven                                                                            |
| blt-version          | 8.9.12                                                                           |
| bootstrap            | Successful                                                                       |
| config-sync          | /var/www/gigaclear/config/default                                                |
| db-driver            | mysql                                                                            |
| db-hostname          | localhost                                                                        |
| db-name              | drupal                                                                           |
| db-password          | drupal                                                                           |
| db-port              | 3306                                                                             |
| db-status            | Connected                                                                        |
| db-username          | drupal                                                                           |
| dev-desktop-enabled  | false                                                                            |
| drupal-settings-file | sites/default/settings.php                                                       |
| drupal-version       | 8.4.4                                                                            |
| drupal-vm-enabled    | true                                                                             |
| drush-alias-files    | /var/www/gigaclear/docroot/../drush/site-aliases/aliases.drushrc.php             |
|                      | /var/www/gigaclear/docroot/../drush/site-aliases/example.local.aliases.drushrc.p |
|                      | hp                                                                               |
|                      | /var/www/gigaclear/docroot/../drush/site-aliases/example.acsf.aliases.drushrc.ph |
|                      | p                                                                                |
| drush-conf           | /var/www/gigaclear/drush/drushrc.php                                             |
| drush-script         | /var/www/gigaclear/vendor/drush/drush/drush.php                                  |
| drush-temp           | /tmp                                                                             |
| drush-version        | 8.1.15                                                                           |
| files                | sites/default/files                                                              |
| install-profile      | lightning                                                                        |
| local-drushrc        | /var/www/gigaclear/docroot/sites/default/local.drushrc.php                       |
| local-settings       | /var/www/gigaclear/docroot/sites/default/settings/local.settings.php             |
| modules              | modules                                                                          |
| php-bin              | /usr/bin/php                                                                     |
| php-conf             | /etc/php/5.6/cli/php.ini                                                         |
| php-mysql            | /var/run/mysqld/mysqld.sock                                                      |
| php-os               | Linux                                                                            |
| private              | /var/www/gigaclear/files-private                                                 |
| root                 | /var/www/gigaclear/docroot                                                       |
| shell                |                                                                                  |
| site                 | sites/default                                                                    |
| temp                 | /tmp                                                                             |
| theme                | bartik                                                                           |
| themes               | themes                                                                           |
| uri                  | http://local.wo_gigaclear.com                                                    |
| user                 |                                                                                  |
+----------------------+----------------------------------------------------------------------------------+
+--------------------------+----------------------------------------------------------------------------------+
| Check                    | Outcome                                                                          |
+--------------------------+----------------------------------------------------------------------------------+
| checkDocrootExists       | Found docroot.                                                                   |
| checkCoreExists          | Drupal core exists                                                               |
| checkSettingsFile        | BLT settings are included in settings file.                                      |
| checkLocalSettingsFile   | Found your local settings file.                                                  |
| checkLocalDrushFile      | Found your local drush settings file.                                            |
| checkUri                 | $options['uri'] is set.                                                          |
| checkUriResponse         | Did not get a response from http://local.wo_gigaclear.com                        |
|                          |                                                                                  |
|                          | Is your *AMP stack running?                                                      |
|                          | Is your web server configured to serve this URI from /var/www/gigaclear/docroot? |
|                          | Is $options['uri'] set correctly in                                              |
|                          | /var/www/gigaclear/docroot/sites/default/local.drushrc.php?                      |
|                          |                                                                                  |
|                          | To generate settings files and install Drupal, run `blt local:setup`             |
|                          |                                                                                  |
| checkFileSystem:%files   | Public files directory is writable.                                              |
| checkFileSystem:%private | Private files directory is writable.                                             |
| checkFileSystem:%temp    | Temporary files directory is writable.                                           |
| checkDbConnection        | Connected to database.                                                           |
| checkDrupalBootstrapped  | Bootstrapped Drupal via drush.                                                   |
| checkDrupalInstalled     | Drupal is installed.                                                             |
| checkCaching:page        | Drupal cache is disabled.                                                        |
| checkCaching:css         | CSS preprocessing is disabled.                                                   |
| checkCaching:js          | JS preprocessing is disabled.                                                    |
| checkComposer:require    | acquia/blt is in composer.json's require object.                                 |
| checkBehat:exists        | Behat local settings file exists.                                                |
| checkBehat:root          | Behat drupal_root is set correctly for Drupal VM.                                |
| checkBehat:uri           | base_url in tests/behat/local.yml does not match the site URI.                   |
|                          |   Behat base_url is set to "http://local.gigaclear.com".                         |
|                          |   Drush site URI is set to http://local.wo_gigaclear.com.                        |
| checkProjectYml:keys     | project.yml has no deprecated keys.                                              |
| checkDrupalVm:alias      | drush.aliases.local exists your drush aliases file.                              |
| checkDrupalVm            | Drupal VM is configured correctly.                                               |
| checkPhpDateTimezone     | PHP setting for date.timezone is correctly set                                   |
+--------------------------+----------------------------------------------------------------------------------+
```

I would like to get a Drupal 8 multisite setup in BLT, I already have a working VM and default Drupal site, but am unsure how to go about getting the extra DB and vhost.  The documentation says to just add them to the config.yml file, but I am unsure how they should be nested, and under what group name I should use.  Also, what a `vagrant provision` be run then, or would I need to completely destroy and rebuild?

Would it be possible to get an example added to the documentation to show how to do this?  I am happy to help by writing up an example if we can get it working on my local environment.

Thanks for your help,
Dan

**Comment by grasmash** [comment]
We've actually added support for this in 9.1.x via `generate:multisite` command. See https://github.com/acquia/blt/pull/1916/files for example config.yml.

**Comment by grasmash** [comment]
Does that answer your question sufficiently?

**Comment by grasmash** [comment]
Closing due to inactivity. I believe the issue has been addressed.

**Comment by retrodans** [comment]
Hi, I have added in the multisites array to my project.yml file and ran vagrant provision, but I when I look at my hosts file, I do not see the new domain listed (which is my first indicator that it isn't working).  I also ssh'd in and looked at the apache vhosts.conf file, and it only lists the original domain.

Do I need to do a complete destroy and rebuild to enable the new multisite functionality?  Also, the commit you linked to also mentions that in Drupal VM (which I believe BLT uses) that it only runs a single site by default.  Do I therefore also need to make these changes?

**Comment by pcsukardi** [comment]
Good day all, i am currently having an issue to get a proper working mutisite configuration using BLT 9 as well. I have added the vhosts and the databases and users to `box/config.yml` ran `vagrant provision` to make the changes. I now ran `blt recipes:mutisite:init` and entered the information into the wizard. This creates a mutisite directory structure in `docroot/sites/example`. When i run `blt setup --define site=example` it is *not* installing into the secondary database i have created in DrupalVM but instead uses the default 'drupal' database. I have tried many different things based on what i have read from the online documentation but cannot seem to get a functioning multisite install with it's own proper drush alias linked to the second database. Am i missing something? Any help would be greatly appreciated, thank you.
